integration-testing/build.gradle.kts

```diff
@@ -63,9 +60,11 @@ val mavenTest by tasks.registering(Test::class) {
 val functionalTest by tasks.registering(Test::class) {
     testClassesDirs = sourceSets["functionalTest"].output.classesDirs
     classpath = sourceSets["functionalTest"].runtimeClasspath
-    
-    systemProperties["kotlinVersion"] = kotlin_version
-    systemProperties["atomicfuVersion"] = atomicfu_snapshot_version
+
+    // the kotlin version used to build the library, which is set in root gradle.properties or overriden by the TC config
+    systemProperties["kotlin_version_for_integration_tests"] = providers.gradleProperty("kotlin_version").orNull
```

Comment:
Is this the right way to extract the `kotlin_version` used to build the library, including the case when the version was overriden by the TC config? 👀